components/settings.js

```diff
@@ -0,0 +1,276 @@
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+          <div>In the storage tab</div>
+        </div>`
+      },
+      dirty: false,
+      data: {}
+    },
+    {
+      name: 'misc',
+      description: 'Miscelleanous settings',
+      render: function (emit) {
+        return html`<div>In the misc tab</div>`
+      },
+      dirty: false,
+      data: {}
+    },
+    {
+      name: 'junk',
+      description: 'Not used for anything',
+      render: function (emit) {
+        return html`<div>In the junk tab</div>`
+      },
+      dirty: false,
+      data: {}
+    }
+  ]
+  this.selected = this.tabs[0].name
+}
+
+Settings.prototype.use = function (emitter) {
+  var self = this
+  emitter.on('settings:toggle', function () {
+    self.toggle()
+    emitter.emit('render')
+  })
+  emitter.on('settings:ontabclick', function (name) {
+    if (self.selected !== name) {
+      console.info('switching to tab (leave for debug purpose)', name)
+      self.selected = name
+      emitter.emit('render')
+    }
+  })
+  emitter.on('settings:apply', function () {
+    console.log('TODO act on settings:apply event')
+  })
+  emitter.on('settings:reset', function () {
+    console.log('TODO act on settings:reset event')
+  })
+}
+
+Settings.prototype.toggle = function () {
+  this.show = !this.show
+}
+
+Settings.prototype.getSelectedTab = function () {
+  var self = this
+  return this.tabs.find(function (tab) { return tab.name === self.selected })
+}
+
+Settings.prototype.render = function (emit) {
+  var self = this
+  if (!self.show) return
+  var cstyle = `width: ${self.width}px;`
+  return html`<div class=${self.containerStyle} style=${cstyle}>
+    ${self.renderTabs(emit)}
+    ${self.renderTabContent(emit)}
+    ${self.renderButtons(emit)}
+  </div>`
+}
+
+Settings.prototype.renderTabs = function (emit) {
+  var self = this
+  var content = this.tabs.map(function (tab, i) {
+    var selected = self.selected === tab.name
+    var cstyle = selected ? `
+      border-top: 1px solid #999;
+      border-right: 1px solid #999;
+      border-left: 1px solid #999;
+      background: rgba(0, 0, 0, 0.3);
+    ` : `
+      border-bottom: 1px solid #999;
+      color: #999;
+    `
+    var name = tab.name
+    return html`<div class=${self.tabStyle} style=${cstyle} onclick=${() => emit('settings:ontabclick', name)}>${name}</div>`
```

Comment:
I know there are some arrow functions in some places like these, but can switch them out later if it's important.